lredis/protocol.lua

```diff
@@ -11,7 +11,10 @@ end
 -- Requests are always just an array of bulk strings
 local function encode_request(arg)
 	local n = arg.n or #arg
-	assert(n > 0, "need at least one argument")
```

Comment:
invalid input should remain as `asserts`: only runtime/environment inflicted errors should `return nil, err`

Reply:
We **cannot** assert here.  If we do, then there is no choice but to wrap the calling code in a ```pcall()``` to catch it, and that is exactly what we're trying to avoid because of the Lua 5.1 issues.